hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -300,6 +311,19 @@ protected List<PartitionPath> listPartitionPaths(List<String> relativePartitionP
   protected List<PartitionPath> listPartitionPaths(List<String> relativePartitionPaths) {
     List<String> matchedPartitionPaths;
     try {
+      if (isPartitionedTable()) {
+        if (queryType == HoodieTableQueryType.INCREMENTAL && beginInstantTime.isPresent()) {
+          HoodieTimeline timelineAfterBeginInstant = TimelineUtils.getCommitsTimelineAfter(metaClient, beginInstantTime.get(), Option.empty());
```

Comment:
This is finding the range (startTs, endTs] for queries. MergeOnReadIncrementalRelation seems to be doing the same thing. This seems like an easy thing to mess up with incremental so just want to highlight it

Reply:
yes, I agree. but not sure if you are suggesting to do anything here for that